Assets/Scripts/System/StateMachine/States/AttractState.cs

```diff
@@ -6,54 +6,147 @@
     {
-
+        helper.attract.SetActive(false);
     }
 
     override public void OnStateUpdate(Animator animator, AnimatorStateInfo info, int layerIndex)
     {
         if (Input.anyKeyDown)
             animator.SetTrigger("NextState");
 
-        playDelay -= Time.deltaTime;
+        displayTime -= Time.deltaTime;
+
+        //Do stuff when the delay is over
+        if (displayTime < 0)
+        {
+            //When playing video, only switch if video is done
+            if (GetCurrentItem().type == AttractItem.AttractItemType.Video)
+            {
+                if (!GM.Instance.video.player.isPlaying)
+                    ShowNextItem();
+            }
 
-        if (playDelay < 0 && !GM.Instance.video.player.isPlaying)
-            PlayNextVideo();
+            //On text or image, just go to the next one
+            else
+            {
+                ShowNextItem();
+            }
+        }
     }
 
-    private void PlayVideo(int number)
+
+    //STATE FUNCTIONS
+
+    /// <summary>
+    /// This shows the current attract item
+    /// </summary>
+    private void ShowCurrentItem()
     {
-        playDelay = 1;
-        GM.Instance.video.PlayVideo(GM.Instance.data.attractFiles[number], false, true);
+        if (GetCurrentItem() != null)
+        {
+            switch (GetCurrentItem().type)
+            {
+                case AttractItem.AttractItemType.Image:
+                    DisplayImage();
+                    return;
+
+                case AttractItem.AttractItemType.Text:
+                    DisplayText();
+                    return;
+
+                case AttractItem.AttractItemType.Video:
+                    DisplayVideo();
+                    return;
+            }
+        }
+
+        else
+        {
+            GM.Instance.logger.Debug("Attract: Item Number does not exist.");
+        }
     }
 
-    private void PlayNextVideo()
+    private void ShowNextItem()
     {
         currentItem++;
 
         if (currentItem > numberOfItems - 1)
             currentItem = 0;
 
-        PlayVideo(currentItem);
+        ShowCurrentItem();
+    }
+
+    private void DisplayImage()
+    {
+        //Turn off unused gameobjects
+        helper.attractText.gameObject.SetActive(false);
+
+        //Play dat background vid
+        GM.Instance.video.PlayBackground();
+
+        //Display the image
+        helper.attractImage.gameObject.SetActive(true);
+        helper.attractImage.sprite = GetCurrentItem().sprite;
+        displayTime = GetCurrentItem().displayTime;
+    }
+
+    private void DisplayVideo()
+    {
+        //Turn off unused GameObjects
+        helper.attractImage.gameObject.SetActive(false);
+        helper.attractText.gameObject.SetActive(false);
+
+        //Play the video!
+        GM.Instance.video.PlayVideo(GetCurrentItem().pathToItem, false, true);
+        displayTime = 1; //just need a second to get the video player running.
```

Comment:
Re: my comment above about `displayTime` vs video length, would it make sense to do

```c#
displayTime = GM.Instance.video.Length; # or whatever the property/method name would be
```

Reply:
I don't think there's a way to do that with VideoPlayer playing from a url instead of a VideoClip.  You can't get a VideoClip using WWW like AudioClips so I use urls, and this method checks for when the VideoPlayer is not playing, but gives it a second to make sure the player is running.  I feel a little more comfortable using this method than trying to run a second timer alongside the VideoPlayer.

Reply:
Ah ok I wondered if something like that would be the case. Whatever works :+1:

Reply:
Don't get me wrong, I'd love to be able to just get the length hah 😄